core/common/lib/keys-lib/src/main/java/org/eclipse/edc/keys/LocalPublicKeyServiceImpl.java

```diff
@@ -57,16 +61,12 @@ private Optional<PublicKey> resolveFromCache(String id) {
     }
 
     private Result<PublicKey> parseKey(String encodedKey) {
-        return registry.parse(encodedKey).compose(pk -> {
+        return registry.parsePublic(encodedKey).compose(pk -> {
             if (pk instanceof PublicKey publicKey) {
                 return Result.success(publicKey);
             } else {
                 return Result.failure("The specified resource did not contain public key material.");
             }
```

Comment:
given that the method is called `parsePublic` could it handle this validation logic internally and return a `PublicKey` directly?